generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/TypeRegistryConfiguration.java

```diff
@@ -123,6 +124,19 @@ private void defineGenerationMode(TypeRegistry tr) {
             transformer));
     }
 
+    private void defineAtomicConstraintType(TypeRegistry tr) {
+        Transformer<AtomicConstraintType> transformer = strategyString ->
+            Arrays.stream(AtomicConstraintType.values())
+                .filter(val -> val.toString().equalsIgnoreCase(strategyString))
+                .findFirst().orElse(AtomicConstraintType.ISOFTYPE);
+
+        tr.defineParameterType(new ParameterType<>(
+            "atomicConstraintType",
+            "(.*)$",
```

Comment:
As per simon's remark earlier it's the $ here which forces it to only bind at the end of the sentence.